test/parallel/test-abortsignal-drop-settled-signals.mjs

```diff
@@ -148,10 +148,16 @@ it('drops settled dependant signals when signal is composite', (t, done) => {
     t.assert.strictEqual(controllers[1].signal[kDependantSignals].size, 1);
 
     setImmediate(() => {
-      t.assert.strictEqual(controllers[0].signal[kDependantSignals].size, 0);
-      t.assert.strictEqual(controllers[1].signal[kDependantSignals].size, 0);
+      global.gc();
 
-      done();
+      setImmediate(() => {
+        global.gc();
```

Comment:
I plan to abstract these multiple GC calls into a function

Reply:
Running tests one more time with `tools/test.py --repeat 9999`

Reply:
I wonder if doing `await global.gc({ async: true })` may give a better result.

Reply:
Interesting. I can check this.

Reply:
That's what they do in https://github.com/nodejs/node/blob/main/deps/v8/test/mjsunit/harmony/weakrefs/weak-cell-basics.js
But they also use `--noincremental-marking`. That might be important too...

@joyeecheung Do you know (or know who knows) how to reliably trigger garbage collection?

Reply:
Is there a reference document for `global.gc`?

Reply:
Not that I know of. There's some documentation in https://github.com/nodejs/node/blob/80c0055e38b9c79b27a2c3b7da3b62e8d0f5e435/deps/v8/src/extensions/gc-extension.h#L19-L52

Reply:
Thanks for all your support here.

Reply:
Frequent use case 2 is:
>  // 2. Test that certain objects indeed are reclaimed: 
 //   `await gc({type:'major', execution:'async'})`

This seems to correspond to our use case.

Reply:
Great. I will do these experiments.

Reply:
The way this PR is now might fix the flakiness. So if it's blocking somebody, I'm okay opening another PR to improve the test using these references.

Whatever you guys decide I'm okay with.

Reply:
There are already some utilities in `test/common/gc.js` and I think the function added here isn't too different from https://github.com/nodejs/node/blob/0b6b2a4dc7afcabbdd35070436440bc8c31e4770/test/common/gc.js#L47 - which has already proven to be unreliable and prone to flakes, by the way, especially when there are ephemeron gc involved (e.g. use of WeakRef/WeakRef/FinalizationRegistry in the graph). `checkIfCollectableByCounting` in there would be slightly more reliable though I'd say if the test relies on GC, there is no way to make it absolutely reliable, the V8 utility you see only ensures that a major GC is performed. That does not mean V8 would be keen on removing any thing after it notices it's unreachable.

Reply:
@joyeecheung would you say that limiting the old-space and heap size would help to increase memory pressure?

Reply:
never mind. we wouldn't know how much is sufficient.